src/Credentials/GCECredentials.php

```diff
@@ -110,6 +113,21 @@ class GCECredentials extends CredentialsLoader implements
      */
     private const GKE_PRODUCT_NAME_FILE = '/sys/class/dmi/id/product_name';
 
+    /**
+     * The Windows Registry key path to the product name
+     */
+    private const REGISTRY_KEY_PATH = 'HKEY_LOCAL_MACHINE\\SYSTEM\\HardwareConfig\\Current\\';
+
+    /**
+     * The Windows registry key name for the product name
+     */
+    private const REGISTRY_KEY_NAME = 'SystemProductName';
+
+    /**
+     * The Name of the product expected from the windows registry
+     */
+    private const WINDOWS_PRODUCT_NAME = 'Google';
```

Comment:
We should either just use the hardcoded string `Google` in the function, or we should rename this to be OS-agnostic (e.g. `GCE_PRODUCT_PREFIX`) and use it in `detectResidencyLinux` as well.